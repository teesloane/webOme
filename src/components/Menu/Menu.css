.Menu {
  display: flex;
  /* background: #333; */
  background: rgb(65, 65, 65);
  color: white;
  flex-direction: column;
  max-width: 250px;
  flex: 1;
  transition: all 0.3s ease-in-out;
  border-right: 1px solid #333;
}

.Menu--hide {
  margin-left: calc(-250px - 4rem);
}

.Menu-section {
  display: flex;
  flex-direction: column;
  padding: 12px 24px;
  background: rgb(65, 65, 65);
  min-height: 70px;
  border-bottom: 2px solid #333;
}

.Menu-octave-box {
  display: flex;
  flex: 1;
  justify-content: space-between;
  align-items: center;
  text-transform: uppercase;
  padding-top: 16px;
  padding-bottom: 16px;
}

.Menu-octave-btn {
  background: #555;
  color: #dfdfdf;
  outline: none;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  font-size: 16px;
  line-height: 0;
  padding-bottom: 4px;
  cursor: pointer;
  border: 2px solid white;
  box-shadow: 1px 1px 3px rgba(0, 0, 0, 0.3);
}

.Menu-octave-btn:active {
  box-shadow: none;
}
