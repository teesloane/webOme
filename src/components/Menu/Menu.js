import React, { PropTypes } from "react";
import { observer, inject } from "mobx-react";
import InputNumber from "../UiKit/InputNumber/InputNumber";
import InputSelect from "../UiKit/InputSelect/InputSelect";
import { KEYS_SELECTOR, SCALES, GRID } from "../../music_constants";
import "./Menu.css";

const Menu = function Menu(props) {
  // midi loads async, so wait until it's loaded until returning.
  // TODO: replace this with proper async notifier / spinner / alert.
  const { UiStore, OmeStore } = props;
  if (!OmeStore.midi) return null;

  return (
    <main className={`Menu ${UiStore.menuOpen ? "" : "Menu--hide"}`}>

      {/*Section: Tempo, Grid */}
      <section className="Menu-section">
        <InputNumber className="input-black" label="Tempo" max="240" min="10" />
      </section>

      <section className="Menu-section">
        <InputSelect
          className="Select-custom"
          name="Grid Resolution"
          clearable={false}
          value={{ label: `1/${OmeStore.grid}`, value: OmeStore.grid }}
          options={GRID}
          onChange={OmeStore.selectGrid}
        />
      </section>

      {/*Section: Scale, Key, Custom Note Mode */}
      <section className="Menu-section">
        {/* Key */}
        <InputSelect
          className="Select-custom"
          name="Key"
          options={KEYS_SELECTOR}
          value={OmeStore.selectedKey}
          onChange={OmeStore.selectKey}
          clearable={false}
          deleteRemoves={false}
        />
      </section>

      {/* Scales */}

      <section className="Menu-section">
        <InputSelect
          className="Select-custom"
          name="Scale"
          options={SCALES}
          onChange={OmeStore.selectScale}
          value={OmeStore.selectedScale}
          clearable={false}
          deleteRemoves={false}
        />
      </section>

      {/* Octave +/- */}
      <section className="Menu-section">
        <section className="Menu-octave-box">
          <button className="Menu-octave-btn" onClick={OmeStore.decrementOctave}>-</button>
          <div>Octave: {OmeStore.octave}</div>
          <button className="Menu-octave-btn" onClick={OmeStore.incrementOctave}>+</button>
        </section>
      </section>

      {/*Section: Midi Setup */}
      {OmeStore.selectedMidiOut
        ? <section className="Menu-section">
            <InputSelect
              className="Select-custom"
              name="Midi Output"
              options={OmeStore.createSelectorOptions(
                OmeStore.midiOutputs,
                "name"
              )}
              value={{
                label: OmeStore.selectedMidiOut.name,
                value: OmeStore.selectedMidiOut
              }}
              onChange={OmeStore.selectMidiDevice}
              clearable={false}
              deleteRemoves={false}
            />
          </section>
        : <section className="Menu-section">
            <InputSelect name="Midi Output" placeholder="No Midi Devices Available" disabled />
          </section>}

    </main>
  );
};

Menu.propTypes = {
  isOpen: PropTypes.bool
};

export default inject("OmeStore", "UiStore")(observer(Menu));
