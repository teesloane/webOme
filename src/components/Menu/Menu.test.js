import Menu from './Menu';

describe('Menu Component', function() {
  beforeEach(function() {
    this.UiStore = {};
    this.OmeStore = {};
  })

  it('renders', () => {
    shallow(<Menu UiStore={this.UiStore} OmeStore={this.OmeStore}/>);
  })

  it('matches snapshot', function() {
    const component = shallow(<Menu UiStore={this.UiStore} OmeStore={this.OmeStore}/>);
    expect(component).toMatchSnapshot();
  });
})