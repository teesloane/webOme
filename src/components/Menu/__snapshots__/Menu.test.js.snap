exports[`Menu Component matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": null,
  "nodes": Array [
    null,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Menu />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance":  {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "render": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": null,
        "_debugID": 4,
        "_renderedOutput": null,
      },
      "_renderedNodeType": 2,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Menu />,
}
`;
